### PR TITLE
Restrict bulk assign to admins of org

### PR DESCRIPTION
**Pull request by hschallhorn**
Bumps #14124

### Description
Only allows admins of an org to bulk assign tasks

### Acceptance Criteria
- [x] Only admin users of organizations are can see bulk assign button
- [x] Only admin users are able to bulk assign tasks to themselves or another team member

### Testing Plan
1. Sign in as BVATWARNER and go to /organizations/hearings-management
1. Confirm no bulk assign button
1. Make BVATWARNER an admin
   ```ruby
   OrganizationsUser.make_user_admin(User.find_by(css_id: "BVATWARNER"), HearingsManagement.singleton)
   ```
1. Ensure can complete bulk assign

### User Facing Changes
 - [x] Only admins can see the bulk assign button

 ADMIN|NON ADMIN
 ---|---
Screen Shot 2020-06-10 at 5 56 47 PM|Screen Shot 2020-06-10 at 5 56 34 PM

### CODE REVIEW

File: app/controllers/organizations/tasks_controller.rb [hunk 1]
```diff
@@ -9,7 +9,8 @@ def index
       organization_name: organization.name,
       id: organization.id,
       is_vso: organization.is_a?(::Representative),
-      queue_config: QueueConfig.new(assignee: organization).to_hash
+      queue_config: QueueConfig.new(assignee: organization).to_hash,
+      user_can_bulk_assign: organization.admins.include?(current_user)
```
Comment (hschallhorn) [inline app/controllers/organizations/tasks_controller.rb:13]: Restrict on the front end

File: app/workflows/bulk_task_assignment.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ class BulkTaskAssignment
   validate :organization_exists
   validate :regional_office_is_valid
   validate :assigned_to_part_of_organization
-  validate :assigned_by_part_of_organization
+  validate :assigned_by_admin_of_organization
```
Comment (hschallhorn) [inline app/workflows/bulk_task_assignment.rb:12]: Restrict on the front end

File: app/workflows/bulk_task_assignment.rb [hunk 3]
```diff
@@ -92,10 +92,10 @@ def assigned_to_part_of_organization
     errors.add(:assigned_to, "does not belong to organization with url #{organization_url}")
   end
 
-  def assigned_by_part_of_organization
-    return if organization&.users&.include?(assigned_by)
+  def assigned_by_admin_of_organization
+    return if organization&.admins&.include?(assigned_by)
```
Comment (hschallhorn) [inline app/workflows/bulk_task_assignment.rb:96]: back end real change

File: client/app/queue/QueueTableBuilder.jsx [hunk 4]
```diff
@@ -97,7 +97,7 @@ class QueueTableBuilder extends React.PureComponent {
       label: sprintf(tabConfig.label, totalTaskCount),
       page: <React.Fragment>
         <p className="cf-margin-top-0">{tabConfig.description}</p>
-        { tabConfig.allow_bulk_assign && <BulkAssignButton /> }
+        { this.props.userCanBulkAssign && tabConfig.allow_bulk_assign && <BulkAssignButton /> }
```
Comment (hschallhorn) [inline client/app/queue/QueueTableBuilder.jsx:100]: front end real change

File: spec/workflows/bulk_task_assignment_spec.rb [hunk 5]
```diff
@@ -108,7 +108,7 @@
       let(:task_type) { "UnknownTaskType" }
 
       it "does not bulk assigns tasks" do
-        organization.users << assigned_by
+        OrganizationsUser.make_user_admin(assigned_by, organization)
         organization.users << assigned_to
         bulk_assignment = BulkTaskAssignment.new(params)
         expect(bulk_assignment.valid?).to eq false
```
Comment (yoomlam) [inline spec/workflows/bulk_task_assignment_spec.rb:114]: Could refactor as `shared_examples`?

Comment (hschallhorn) [reply spec/workflows/bulk_task_assignment_spec.rb]: Done and done!

File: spec/workflows/bulk_task_assignment_spec.rb [hunk 6]
```diff
@@ -47,116 +47,73 @@
...
     let(:regional_office) { nil }
 
-    context "when assigned to user does not belong to organization" do
-      it "does not bulk assigns tasks" do
-        organization.users << assigned_by
-        bulk_assignment = BulkTaskAssignment.new(params)
-        expect(bulk_assignment.valid?).to eq false
-        error = ["does not belong to organization with url #{organization.url}"]
-        expect(bulk_assignment.errors[:assigned_to]).to eq error
+    let(:assigner) { assigned_by }
```
Comment (hschallhorn) [inline spec/workflows/bulk_task_assignment_spec.rb:50]: Updated all `organization.users << assigned_by` to `OrganizationsUser.make_user_admin(assigner, organization)` and broke out the invalid test cases into a shard example!

**Pull request merged**